### PR TITLE
Allow additional properties in db stream state

### PR DESCRIPTION
**Pull request by tuliren**
## What
- I am going to introduce a new field in `DbStreamState` for https://github.com/airbytehq/airbyte/issues/14732.
- This PR prepares the main JDBC source connectors for that change by allowing additional properties in the JSON schema.
  - Postgres
  - MySQL
  - MSSQL
- In this way, if anything goes wrong with the new field, there is a version that users can rollback to without breaking existing connectors.

## 🚨 User Impact 🚨
None expected.

### CODE REVIEW

File: airbyte-integrations/connectors/source-relational-db/src/main/resources/db_models/db_models.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ definitions:
           "$ref": "#/definitions/DbStreamState"
   DbStreamState:
     type: object
-    additionalProperties: false
+    additionalProperties: true
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-relational-db/src/main/resources/db_models/db_models.yaml:30]: This is the main change.

**Review by evantahler — APPROVED** [review]
👍 

There are a lot of PRs these days turning `additionalProperties` to true.  Should we do that everywhere?

**Comment by tuliren** [comment]
> There are a lot of PRs these days turning `additionalProperties` to `true`. Should we do that everywhere?

I think so. Anything with `additionalProperties = false` is not forward compatible, thus preventing users from downgrading. I really don't see the harm for setting `additionalProperties` to `true` universally.

**Comment by tuliren** [comment]
Manually cancelled all publication because there is a build failure.

**Pull request merged**